src/commands/react-native/xcode.ts

```diff
@@ -354,15 +369,38 @@ export class XCodeCommand extends Command {
    * return false if the React Native version is high enough.
    */
   private shouldComposeHermesSourcemaps = (): boolean => {
+    /**
+     * This env variable is empty by default.
+     * Before RN 0.70, it had to be set to `true` for Hermes to be used.
+     * Since RN 0.70, Hermes is enabled even if it is empty.
+     */
     if (process.env.USE_HERMES) {
       return true
     }
-    if (process.env.HERMES_CLI_PATH) {
+
+    /**
+     * Check if hermes pod is present in pods.
+     * This is the check used until RN 0.70, but the architecture of the pod might change,
+     * so it's best not to rely on it to detect if Hermes is disabled.
+     */
+    if (existsSync(`${process.env.PODS_ROOT}/hermes-engine/destroot/bin/hermesc`)) {
       return true
     }
 
-    // Check if hermes pod is present
-    return existsSync(`${process.env.PODS_ROOT}/hermes-engine/destroot/bin/hermesc`)
+    /**
+     * Checks if Hermes is in the Podfile.lock.
+     * This is the most recent check for detecting Hermes in the `react-native-xcode.sh` script:
+     * https://github.com/facebook/react-native/commit/8745a148b6d8358702b5300d73f4686c3aedb413
+     *
+     * If the Podfile.lock cannot be found, we assume Hermes is not enabled
```

Comment:
Is there a situation where you could not have a Podfile.lock but still have a valid build, even in older versions of React Native? I'm wondering if this should at least give a warning if not an outright error.